proof_of_stake/src/pos_queries.rs

```diff
@@ -208,22 +206,11 @@ where
     pub fn is_deciding_offset_within_epoch(self, height_offset: u64) -> bool {
         let current_decision_height = self.get_current_decision_height();
 
-        // NOTE: the first stored height in `fst_block_heights_of_each_epoch`
-        // is 0, because of a bug (should be 1), so this code needs to
-        // handle that case
-        //
-        // we can remove this check once that's fixed
-        if self.wl_storage.storage.get_current_epoch().0 == Epoch(0) {
-            let height_offset_within_epoch = BlockHeight(1 + height_offset);
-            return current_decision_height == height_offset_within_epoch;
-        }
-
-        let fst_heights_of_each_epoch = self
-            .wl_storage
+        let pred_epochs = self
```

Comment:
@sug0 I rewrote this module to use `StorageRead` instead of the full `WlStorage`. I removed this hack as this was fixed in https://github.com/anoma/namada/pull/1993 - I had to update the test in https://github.com/anoma/namada/pull/2312/commits/0354bbec89d1947c1e80bf7c4255031b8bc7d766, pls check if that's right

Reply:
Ah, the `init_chain` still initializes `pred_epochs` with `BlockHeight(0)`. I'll put the hack back in for the refactor and try to fix it in a follow-up